src/server/udp/session/session.go

```diff
@@ -69,54 +75,58 @@ func NewSession(clientAddr *net.UDPAddr, conn *net.UDPConn, backend core.Backend
 			tC = t.C
 		}
 
+		// wait for; the idle timer, output to send to the backend, or a close request
 		for {
 			select {
 
 			case <-tC:
-				s.Close()
-			case buf := <-s.out:
+				s.BackendClose()
+
+			case sendBufPool := <-s.out:
```

Comment:
we're getting a slice from "bufPool", and sending it.
*shrug*
lets make it better!
it probably starts by updating;
var bufPool = sync.Pool{
because bufPool isnt particularly descriptive... maybe it's a packetBufferPool